### PR TITLE
Introduce new Key Manager endpoints

### PR DESCRIPTION
**Pull request by joaquim-verges**
Adds the new Ethereum Key manager endpoints ( as defined in https://github.com/ethereum/keymanager-APIs) in the OpenAPI spec.

Also adds a pass through (for now) security handler to the OpenAPI router factory (required by Vertx)

This is one of the tasks of the Key manager API integration discussed in #439

### CODE REVIEW

File: core/src/main/resources/openapi-specs/eth2/keymanager/paths/keystores.yaml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                        - error: Any other status different to the above: decrypting error, I/O errors, etc.
+                      enum:
+                        - imported
+                        - duplicate
+                        - error
+                      example: imported
+                    message:
+                      type: string
+                      description: error message if status == error
+    "401":
```
Comment (jframe) [inline core/src/main/resources/openapi-specs/eth2/keymanager/paths/keystores.yaml:104]: This is missing the 400 bad response from the keymanager openapi spec.

File: core/src/main/resources/openapi-specs/eth2/keymanager/paths/keystores.yaml [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                        - not_active
+                        - not_found
+                        - error
+                      example: deleted
+                    message:
+                      type: string
+                      description: error message if status == error
+              slashing_protection:
+                $ref: "../schemas.yaml#/components/schemas/SlashingProtectionData"
+    "401":
```
Comment (jframe) [inline core/src/main/resources/openapi-specs/eth2/keymanager/paths/keystores.yaml:175]: Missing 400 bad request response

File: core/src/main/resources/openapi-specs/eth2/keymanager/schemas.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      example: '{"metadata":{"interchange_format_version":"5","genesis_validators_root":"0xcf8e0d4e9587369b2301d0790347320302cc0943d5a1884560367e8208d920f2"},"data":[{"pubkey":"0x93247f2209abcacf57b75a51dafae777f9dd38bc7053d1af526f220a7489a6d3a2753e5f3e8b1cfe39b56f43611df74a","signed_blocks":[],"signed_attestations":[]}]}'
+
+    ErrorResponse:
+      type: object
+      required: [message]
+      properties:
+        message:
+          description: "Message describing error"
+          type: string
+          example: "Internal server error"
```
Comment (jframe) [inline core/src/main/resources/openapi-specs/eth2/keymanager/schemas.yaml:35]: In the spec it has the values instead
```
          description: "Detailed error message"
          type: string
          example: "description of the error that occurred"
```

File: core/src/main/resources/openapi-specs/eth2/keymanager/schemas.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+  responses:
+    Unauthorized:
```
Comment (jframe) [inline core/src/main/resources/openapi-specs/eth2/keymanager/schemas.yaml:38]: Missing the BadRequest response
```
    BadRequest:
      description: "Bad request. Request was malformed and could not be processed"
      content:
        application/json:
          schema:
            $ref: "#/components/schemas/ErrorResponse"
```

File: core/src/main/resources/openapi-specs/eth2/keymanager/schemas.yaml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    Forbidden:
+      description: "Forbidden, a token is found but is invalid"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ErrorResponse"
+
+    InternalError:
+      description: "Beacon node internal error."
```
Comment (jframe) [inline core/src/main/resources/openapi-specs/eth2/keymanager/schemas.yaml:53]: The description here is different from the spec. In the spec it has `      description: "Internal server error. The server encountered an unexpected error indicative of
                    a serious fault in the system, or a bug."`

File: core/src/main/java/tech/pegasys/web3signer/core/Runner.java [hunk 6]
```diff
@@ -123,6 +123,20 @@ public void run() {
       if (config.isAccessLogsEnabled()) {
         routerFactory.addGlobalHandler(LoggerHandler.create(LoggerFormat.DEFAULT));
       }
+
+      // security handler for keymanager endpoints
+      routerFactory.addSecurityHandler(
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/Runner.java:128]: Can this be moved into the Eth2Runner as this only applies for Eth2

File: core/src/main/java/tech/pegasys/web3signer/core/Runner.java [hunk 7]
```diff
@@ -123,6 +123,20 @@ public void run() {
...
+          "bearerAuth",
+          context -> {
+            // TODO Auth token security logic
+            final boolean autorized = true;
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/Runner.java:132]: nit: authorized instead of autorized

Comment (joaquim-verges) [reply core/src/main/java/tech/pegasys/web3signer/core/Runner.java]: woops

File: core/src/main/java/tech/pegasys/web3signer/core/Runner.java (hunk 6 above)
Comment (joaquim-verges) [reply core/src/main/java/tech/pegasys/web3signer/core/Runner.java]: done

File: core/src/main/resources/openapi-specs/eth2/keymanager/schemas.yaml (hunk 5 above)
Comment (joaquim-verges) [reply core/src/main/resources/openapi-specs/eth2/keymanager/schemas.yaml]: updated everything to the latest spec 👍

File: core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java [hunk 8]
```diff
@@ -129,6 +129,19 @@ private void registerEth2Routes(
       final Optional<SlashingProtection> slashingProtection) {
     final ObjectMapper objectMapper = SigningObjectMapperFactory.createObjectMapper();
 
+    // security handler for keymanager endpoints
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java:132]: nit: Would be better if adding this was conditional on the key manager api being enabled

**Review by jframe — APPROVED** [review]

File: core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java (hunk 8 above)
Comment (joaquim-verges) [reply core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java:132]: unfortunately Vertx requires this at parsing time, as soon as it sees an auth'd endpoint in the spec it goes look for the a security handler and crashes if none is found.

**Pull request merged**